### ISSUE TITLE
if u can add a running apk or upload it to google play so we can try it 

### ISSUE THREAD

**Issue by MahmoudMabrok** [AmrDeveloper/CodeView#1]

**Comment by AmrDeveloper** [comment]
Thanks @MahmoudMabrok 

For now, you can run the example app 
that contains examples for 3 Languages Java, Python, and GoLang with 4 Themes for every one 
and also you can check the project list

Later I will add more examples that show how error lines work and cover other features
and developers also can add their projects

**Comment by MahmoudMabrok** [comment]
@AmrDeveloper  good work

**Comment by ansh2166** [comment]
@AmrDeveloper can u add HTML in examples too? 👍

**Comment by AmrDeveloper** [comment]
@ansh2166 Hello bro, thanks for your suggestion, I will add a full example that will cover most of the syntax of HTML but for now you can start with just 4 patterns and you will get this result

```java
private static final Pattern PATTERN_NUMBERS    = Pattern.compile("\\b(\\d*[.]?\\d+)\\b");
private static final Pattern PATTERN_STRING     = Pattern.compile("\".*\"");
private static final Pattern PATTERN_TAGS       = Pattern.compile("<[/]?\\s*(\\w+)[>]?");
private static final Pattern PATTERN_ATTRIBUTE  = Pattern.compile("[a-zA-Z]+=");
```
![codeview_html](https://user-images.githubusercontent.com/23631699/142070516-7103a2e5-0dfb-48aa-afc5-e58777a7705c.PNG)

**Comment by ansh2166** [comment]
@AmrDeveloper i dont see line number in my code editor view, any thing I needa do?

**Comment by ansh2166** [comment]
I am pretty interested in library, I would love to see search fuctions, it will highlight that specific piece of code and jump to that line

**Comment by ansh2166** [comment]
![image](https://user-images.githubusercontent.com/72782632/144752860-44831eac-d7c8-4a11-8573-654d148b7843.png)
a little issue, inner content of String is highlighted too if its filtered, any solutions?

**Comment by AmrDeveloper** [comment]
Hello @ansh2166,

The problem of highlighted string content is because the order of the patterns try to change the order of keyword with the string patterns, 

Please keep the issues separated, so create one issue for each problem with more details so we can discuss each one and track them

Thanks

**Comment by AddyDevz** [comment]
Thanks for this library and that regex code, i just made an app using it

Light mode :
![image](https://user-images.githubusercontent.com/98032756/150158816-49632b28-f87a-4e60-a282-9df115de4e33.png)

Dark mode : 
![image](https://user-images.githubusercontent.com/98032756/150158894-e4f945c2-d337-4f91-829e-bd5fdd603d9e.png)

**Comment by AddyDevz** [comment]
BTW, can we add custom font?

**Comment by AmrDeveloper** [comment]
> Thanks for this library and that regex code, I just made an app using it

You're welcome, I am very happy to see your app, good work

**Comment by AmrDeveloper** [comment]
> BTW, can we add custom font?

Sure you can add it easily, remember that CodeView is based on `MultiAutoCompleteTextView` class, so you can change the font easily using `setTypeface` method like any normal TextView

**Comment by AddyDevz** [comment]
Okay, thanks for your cool library

**Comment by AmrDeveloper** [comment]
You're most welcome, enjoy 😄

**Comment by AddyDevz** [comment]
Would be nice if you use it with `Jetbrains Mono` font, results are damn cool

![image](https://user-images.githubusercontent.com/98032756/150368602-df8e5021-ccc3-427e-8b88-f13ffaf75d2b.png)

**Comment by AmrDeveloper** [comment]
@AddyDevz Yes it looks very cool I will use this font in the example app 😄 

Thank you for the suggestion and demo screenshot

**Comment by rudrathedev** [comment]
I'm not so pro in regex patterns, is there any possibility to detect css stuff start, more precisely like `.mydiv {`

**Comment by AmrDeveloper** [comment]
> I'm not so pro in regex patterns, is there any possibility to detect css stuff start, more precisely like `.mydiv {`

Hello @rudrathedev 

You can use some regex like `[.][a-zA-Z]+ [{]` to match `.mydiv {`

You can learn and test your regex again text input easily using this cool website [Regex101.com](https://regex101.com/)

BTW: soon there will be more examples on how to config some other popular languages 😄

**Comment by ansh2166** [comment]
Perhaps, this can be closed 🤔

**Comment by AmrDeveloper** [comment]
@ansh2166 We can keep it opened for now because it has a lot of good Q&As

**Comment by m-anshuman2166** [comment]
https://bit.ly/3JxSOF0

An app using codeView 😁
